#script to align H.zea WGRS reads with bowtie2
#12/13/18
#MF

cd /media/megan/"New Volume"

mkdir Hzea_WGRS_Bowtie2_output

#end-to-end alignments with highest sensitivity
##2002 samples first
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_1_S17_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_1_S17_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_01_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_1_S17_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_1_S17_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_01_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_2_S18_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_2_S18_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_02_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_2_S18_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_2_S18_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_02_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_3_S19_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_3_S19_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_03_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_3_S19_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_3_S19_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_03_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_4_S20_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_4_S20_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_04_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_4_S20_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_4_S20_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_04_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_5_S21_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_5_S21_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_05_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_5_S21_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_5_S21_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_05_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_6_S22_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_6_S22_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_06_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_6_S22_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_6_S22_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_06_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_7_S23_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_7_S23_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_07_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_7_S23_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_7_S23_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_07_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_10_S24_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_10_S24_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_10_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_10_S24_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2002_10_S24_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2002_10_unpaired.sam

#2017 samples

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_1_S25_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_1_S25_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_01_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_1_S25_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_1_S25_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_01_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_3_S26_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_3_S26_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_03_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_3_S26_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_3_S26_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_03_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_5_S27_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_5_S27_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_05_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_5_S27_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_5_S27_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_05_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_6_S28_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_6_S28_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_06_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_6_S28_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_6_S28_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_06_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_7_S29_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_7_S29_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_07_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_7_S29_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_7_S29_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_07_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_8_S30_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_8_S30_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_08_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_8_S30_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_8_S30_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_08_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_9_S31_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_9_S31_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_09_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_9_S31_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_9_S31_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_09_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_10_S32_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_10_S32_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_10_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_10_S32_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed/2017_10_S32_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_2017_10_unpaired.sam


#Bt Samples

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_1_S1_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_1_S1_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_1_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_1_S1_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_1_S1_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_1_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_2_S2_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_2_S2_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_2_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_2_S2_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_2_S2_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_2_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_3_S3_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_3_S3_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_3_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_3_S3_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_3_S3_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_3_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_4_S4_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_4_S4_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_4_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_4_S4_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_4_S4_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_4_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_5_S5_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_5_S5_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_5_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_5_S5_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_5_S5_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_5_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_6_S6_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_6_S6_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_6_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_6_S6_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_6_S6_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_6_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_7_S7_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_7_S7_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_7_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_7_S7_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_7_S7_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_7_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_8_S8_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_8_S8_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_8_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_8_S8_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Bt_8_S8_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Bt_8_unpaired.sam

#Non-Bt

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_1_S9_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_1_S9_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_1_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_1_S9_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_1_S9_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_1_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_2_S10_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_2_S10_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_2_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_2_S10_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_2_S10_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_2_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_3_S11_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_3_S11_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_3_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_3_S11_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_3_S11_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_3_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_4_S12_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_4_S12_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_4_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_4_S12_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_4_S12_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_4_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_5_S13_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_5_S13_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_5_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_5_S13_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_5_S13_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_5_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_6_S14_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_6_S14_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_6_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_6_S14_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_6_S14_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_6_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_7_S15_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_7_S15_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_7_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_7_S15_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_7_S15_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_7_unpaired.sam

bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -1 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_8_S16_forward_paired.fq.gz -2 ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_8_S16_reverse_paired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_8_paired.sam
bowtie2 -x ./Hz_PopGen_ddRAD_demult/Bowtie_genome_alignments/Hzea_genome/Hzea_genome -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_8_S16_forward_unpaired.fq.gz -U ./Raw_Sequence_Data/H.zea_WGRS_DivelySamples/GouldFritz/trimmed_BtandNonBt/Non-Bt_8_S16_reverse_unpaired.fq.gz --very-sensitive --threads 3 -S ./Hzea_WGRS_Bowtie2_output/HZ_Non-Bt_8_unpaired.sam


