---
output:
  word_document: default
  html_document: default
---

```{r}
library(tidyverse)
```

```{r}
BAP11A1_CL <- read.csv("~/Desktop/UMD/Crosses_Analysis/gemma/BAP11A1_CL_all_snps_4.5.21.csv") %>% filter(allele_frequency > 0.05)

BAP11A1_DD <- read.csv("~/Desktop/UMD/Crosses_Analysis/gemma/BAP11A1_DD_all_snps_4.5.21.csv") %>% filter(allele_frequency > 0.05)

DEO9A1_CL <- read.csv("~/Desktop/UMD/Crosses_Analysis/gemma/DEO9A1_CL_all_snps_4.5.21.csv") %>% filter(allele_frequency > 0.05)

DEO9A1_DD <- read.csv("~/Desktop/UMD/Crosses_Analysis/gemma/DEO9A1_DD_all_snps_4.5.21.csv") %>% filter(allele_frequency > 0.05)

map <- read.table("~/Desktop/UMD/Crosses_Analysis/B.mori/Hzea_superScaf_genome.txt", header = TRUE)
```

## Make dataframes of associated SNPs for all treatments.

```{r}
sig_DEO9A1_CL <-  DEO9A1_CL %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05)

sig_BAP11A1_CL <- BAP11A1_CL %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05)

sig_BAP11A1_DD <- BAP11A1_DD %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05)

sig_DEO9A1_DD <- DEO9A1_DD %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05)
```

## Calculate number of associated SNPs and scaffolds.

```{r}
BAP11A1_DD %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05) %>%
  filter(!scaffold %in% sig_BAP11A1_CL$scaffold) %>%
  summarize(n_snps = n(), n_scaffolds = n_distinct(scaffold))

```

```{r}
BAP11A1_DD %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.00000001) %>%
  filter(allele_frequency > 0.05) %>%
  filter(!scaffold %in% sig_BAP11A1_CL$scaffold) %>%
  summarize(n_snps = n(), n_scaffolds = n_distinct(scaffold))

```

```{r}
DEO9A1_DD %>%
   filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05) %>%
  filter(!scaffold %in% sig_BAP11A1_CL$scaffold) %>%
  summarize(n_snps = n(), n_scaffolds = n_distinct(scaffold))
```

```{r}
DEO9A1_DD %>%
   filter(gamma > 0.01) %>%
  filter(p_lrt < 0.00000001) %>%
  filter(allele_frequency > 0.05) %>%
  filter(!scaffold %in% sig_BAP11A1_CL$scaffold) %>%
  summarize(n_snps = n(), n_scaffolds = n_distinct(scaffold))
```

## Identify scaffolds associated with growth on both DD and CL treatment

```{r}
BAP11A1_DD %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05) %>%
  filter(scaffold %in% sig_BAP11A1_CL$scaffold) %>%
  group_by(scaffold) %>%
  summarize(n_snps = n())
```

```{r}
DEO9A1_DD %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05) %>%
  filter(scaffold %in% sig_BAP11A1_CL$scaffold) %>%
  group_by(scaffold) %>%
  summarize(n_snps = n())
```

## Identify linkage groups with growth associated SNPs

```{r}
BAP11A1_DD <- merge(BAP11A1_DD, map, by.x = "scaffold", by.y = "Scaf", all.x = TRUE)
BAP11A1_CL <- merge(BAP11A1_CL, map, by.x = "scaffold", by.y = "Scaf", all.x = TRUE)

DEO9A1_DD <- merge(DEO9A1_DD, map, by.x = "scaffold", by.y = "Scaf", all.x = TRUE)
DEO9A1_CL <- merge(DEO9A1_CL, map, by.x = "scaffold", by.y = "Scaf", all.x = TRUE)
```

```{r}
BAP11A1_DD %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05) %>%
  group_by(Chr) %>%
  summarize(n_snps = n())

```

```{r}
DEO9A1_DD %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05) %>%
  group_by(Chr) %>%
  summarize(n_snps = n())
```

```{r}
BAP11A1_CL %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05) %>%
  group_by(Chr) %>%
  summarize( n= n())

```


```{r}
DEO9A1_CL %>%
  filter(gamma > 0.01) %>%
  filter(p_lrt < 0.01) %>%
  filter(allele_frequency > 0.05) %>%
  group_by(Chr) %>%
  summarize( n= n())

```

## Compare to previously identified candidates

```{r}
names <- c("alp","apn1","apn4","abcA2","abcC2","abcG1","calp","cad2","cad-86C","map4K4","tspan1")
candidate_scaf <- c("KZ117832.1","KZ118301.1","KZ118301.1","KZ118207.1","KZ118297.1","KZ118424.1","KZ117563.1","KZ118195.1","KZ117463.1","KZ118817.1","KZ118424.1")
```

```{r}
for (i in 1:length(candidate_scaf)){
   crom <- map %>% 
     subset(map$Scaf %in% candidate_scaf[i]) %>% 
     select(Chr)
   if (nrow(crom) > 0 ) {
    print(paste(names[i], "is on scaffold", candidate_scaf[i], "and linkage group", crom$Chr )) }
   else{
     print(paste(names[i], "is on scaffold", candidate_scaf[i], "and is not in map")) 
   }
}
```

## Two test criteria 

```{r}
for (i in 1:length(candidate_scaf)){
  sig_markers <- BAP11A1_DD %>% 
    subset(BAP11A1_DD$scaffold %in% candidate_scaf[i]) %>%
    filter(gamma > 0.01) %>%
    filter(p_lrt < 0.01) %>%
    filter(allele_frequency > 0.05)
     
  if (nrow(sig_markers) > 0){
    print(paste(names[i], "is on scaffold", candidate_scaf[i], "with growth associated marker", sig_markers$snp_id))}
  if (!candidate_scaf[i] %in% BAP11A1_DD$scaffold) {
    print(paste("THERE IS NO MARKER ON THE SAME SCAFFOLD AS", names[i], "FOR BAP11A1_DD"))}}

```


```{r}
for (i in 1:length(candidate_scaf)){
  sig_markers <- DEO9A1_DD  %>%  
    subset(sig_DEO9A1_DD$scaffold %in% candidate_scaf[i]) %>%
    filter(gamma > 0.01) %>%
    filter(p_lrt < 0.01) %>% 
    filter(allele_frequency > 0.05)
     
  if (nrow(sig_markers) > 0){
    print(paste(names[i], "is on scaffold", candidate_scaf[i], "with growth associated marker", sig_markers$snp_id))}
  if (!candidate_scaf[i] %in% DEO9A1_DD$scaffold) {
    print(paste("THERE IS NO MARKER ON THE SAME SCAFFOLD AS", names[i], "DEO9A1_DD"))}}

```

```{r}
for (i in 1:length(candidate_scaf)){
  sig_markers <- BAP11A1_CL %>% 
    subset(BAP11A1_CL$scaffold %in% candidate_scaf[i]) %>%
    filter(gamma > 0.01) %>%
    filter(p_lrt < 0.01) %>% 
    filter(allele_frequency > 0.05)
   
  if (nrow(sig_markers) > 0){
    print(paste(names[i], "is on scaffold", candidate_scaf[i], "with growth associated marker", sig_markers$snp_id))}
  if (!candidate_scaf[i] %in% BAP11A1_CL$scaffold) {
    print(paste("THERE IS NO MARKER ON THE SAME SCAFFOLD AS", names[i], "BAP11A1_CL"))}}

```


```{r}
for (i in 1:length(candidate_scaf)){
  sig_markers <- sig_DEO9A1_CL %>% 
    subset(sig_DEO9A1_CL$scaffold %in% candidate_scaf[i]) %>%
    filter(gamma > 0.01) %>%
    filter(p_lrt < 0.01) %>% 
    filter(allele_frequency > 0.05)
    
  if (nrow(sig_markers) > 0){
    print(paste(names[i], "is on scaffold", candidate_scaf[i], "with growth associated marker", sig_markers$snp_id))}
  if (!candidate_scaf[i] %in% DEO9A1_CL$scaffold) {
    print(paste("THERE IS NO MARKER ON THE SAME SCAFFOLD AS", names[i], "DEO9A1_CL"))}}

```

## Relaxed test criteria 

```{r}
for (i in 1:length(candidate_scaf)){
  sig_markers <- BAP11A1_DD %>% 
    subset(BAP11A1_DD$scaffold %in% candidate_scaf[i]) %>%
    filter(p_lrt < 0.01) %>%
    filter(allele_frequency > 0.05)
     
  if (nrow(sig_markers) > 0){
    print(paste(names[i], "is on scaffold", candidate_scaf[i], "with growth associated marker", sig_markers$snp_id))}
  if (!candidate_scaf[i] %in% BAP11A1_DD$scaffold) {
    print(paste("THERE IS NO MARKER ON THE SAME SCAFFOLD AS", names[i], "FOR BAP11A1_DD"))}}

```


```{r}
for (i in 1:length(candidate_scaf)){
  sig_markers <- DEO9A1_DD  %>%  
    subset(DEO9A1_DD$scaffold %in% candidate_scaf[i]) %>%
    filter(p_lrt < 0.01) %>% 
    filter(allele_frequency > 0.05)
     
  if (nrow(sig_markers) > 0){
    print(paste(names[i], "is on scaffold", candidate_scaf[i], "with growth associated marker", sig_markers$snp_id))}
  if (!candidate_scaf[i] %in% DEO9A1_DD$scaffold) {
    print(paste("THERE IS NO MARKER ON THE SAME SCAFFOLD AS", names[i], "DEO9A1_DD"))}}

```


```{r}
for (i in 1:length(candidate_scaf)){
  sig_markers <- BAP11A1_CL %>% 
    subset(BAP11A1_CL$scaffold %in% candidate_scaf[i]) %>%
    filter(p_lrt < 0.01) %>%
    filter(allele_frequency > 0.05)
     
  if (nrow(sig_markers) > 0){
    print(paste(names[i], "is on scaffold", candidate_scaf[i], "with growth associated marker", sig_markers$snp_id))}
  if (!candidate_scaf[i] %in% BAP11A1_DD$scaffold) {
    print(paste("THERE IS NO MARKER ON THE SAME SCAFFOLD AS", names[i], "FOR BAP11A1_DD"))}}

```

```{r}
for (i in 1:length(candidate_scaf)){
  sig_markers <- DEO9A1_CL %>% 
    subset(DEO9A1_CL$scaffold %in% candidate_scaf[i]) %>%
    filter(p_lrt < 0.01) %>%
    filter(allele_frequency > 0.05)
     
  if (nrow(sig_markers) > 0){
    print(paste(names[i], "is on scaffold", candidate_scaf[i], "with growth associated marker", sig_markers$snp_id))}
  if (!candidate_scaf[i] %in% BAP11A1_DD$scaffold) {
    print(paste("THERE IS NO MARKER ON THE SAME SCAFFOLD AS", names[i], "FOR DEO9A1_CL"))}}

```


```{r}
BAP11A1_CL %>%
filter(scaffold == "KZ118207.1") %>%
  #filter(radtag_position < 200000)%>%
    filter(p_lrt < 0.01) %>%
    filter(gamma > 0.01) %>%
    filter(allele_frequency > 0.05)


```

